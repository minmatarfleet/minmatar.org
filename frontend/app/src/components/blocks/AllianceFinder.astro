---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import Dialog from '@components/blocks/Dialog.astro';
import Input from '@components/blocks/Input.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
import FixedFluid from '@components/compositions/FixedFluid.astro';
import VerticalCenter from './VerticalCenter.astro';

import alliances from '@json/hosted/alliances.json';

import { get_alliance_logo } from '@helpers/eve_image_server';

interface Props {
    [propName: string]: any;
}
---

<div
    class="[ alliances_finder ]"    
    x-data={`{
        filter: '',
        show_item(el) {
            const name_container = el.querySelector('span')
            const show_item = (
                this.filter != '' && name_container.textContent.toLowerCase().includes(this.filter.toLowerCase())
            )

            return show_item
        }
    }`}
>
    <Dialog
        x-bind:class="(alliances_finder_open ? 'open' : '')"
        x-bind:aria-hidden="alliances_finder_open == false"
        breakout={true}
        class="[ w-full max-w-[500px] ]"
        x-trap="alliances_finder_open"
        x-bind:inert="!alliances_finder_open"
    >
        <Input placeholder={t('find_alliance_placeholder')} x-model="filter" />

        <div class="[ relative ]">
            <Flexblock class="[ alliance-list ][ light-transparency w-full ]" gap='0'>
                {alliances.map((alliance) =>
                    <button
                        type="button"
                        alliance-finder-action="accept"
                        alliance-finder-data={JSON.stringify(alliance)}
                        x-show="show_item($el)"
                        x-on:click="(filter = '')"
                    >
                        <FixedFluid width='64px'>
                            <picture>
                                <img loading="lazy" src={get_alliance_logo(alliance.alliance_id, 64)} height="64" width="64" alt={`${alliance.name} ${t('profile_picture')}`} />
                            </picture>
                            <VerticalCenter>
                                <Flexblock gap='0' class="[ items-start ]">
                                    <span class="[ truncate ]">{alliance.name}</span>
                                    <small>{`${alliance.members_count} ${t('members_count').toLowerCase()}`}</small>
                                </Flexblock>
                            </VerticalCenter>
                        </FixedFluid>
                    </button>
                )}
            </Flexblock>
        </div>
    </Dialog>

    <div
        x-bind:class="(alliances_finder_open ? 'open' : '')"
        alliance-finder-action="close"
        x-on:click="(filter = '')"
    />
</div>

<style lang="scss">
    :global([alliance-finder-action] *) {
        pointer-events: none !important;
    }

    .dialog {
        top: 22%;
    }
    
    .alliance-list {
        position: absolute;
        overflow-y: auto;
        max-height: 320px;
        background-color: #171616;
        border: solid 1px var(--border-color);
        border-top: none;
    }

    [alliance-finder-action="close"] {
        position: fixed;
        inset: 0;
        z-index: var(--backdrop-z-index);
        display: none;

        &.open {
            display: block;
        }
    }

    [alliance-finder-action="accept"] {
        background-color: transparent;
        cursor: pointer;
        border: none;
        padding: 0;

        &:focus-within {
            outline: none;
            background-color: var(--component-background);
        }

        span {
            color: var(--highlight);
        }

        small {
            color: var(--foreground);
        }
        
        &:hover {
            background-color: var(--component-background);
        }
    }
</style>