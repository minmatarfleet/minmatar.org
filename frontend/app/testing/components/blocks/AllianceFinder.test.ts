import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import AllianceFinder from "@components/blocks/AllianceFinder.astro";

test("AllianceFinder defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(AllianceFinder, {});

  expect(result).toMatchSnapshot();
});
