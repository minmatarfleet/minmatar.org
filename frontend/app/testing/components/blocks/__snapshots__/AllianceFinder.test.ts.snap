// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AllianceFinder defaults 1`] = `
"<div class="[ alliances_finder ]" x-data="{
        filter: '',
        show_item(el) {
            const name_container = el.querySelector('span')
            const show_item = (
                this.filter != '' &#38;&#38; name_container.textContent.toLowerCase().includes(this.filter.toLowerCase())
            )

            return show_item
        }
    }" data-astro-cid-e5qsd3eu> <div role="dialog" class="dialog solid-transparency breakout fixed [ w-full max-w-[500px] ]" x-bind:class="(alliances_finder_open ? 'open' : '')" x-bind:aria-hidden="alliances_finder_open == false" x-trap="alliances_finder_open" x-bind:inert="!alliances_finder_open" data-astro-cid-e5qsd3eu="true"> <div class="wrapper" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-m);--wrapper_padding_inline_end: var(--space-m);--wrapper_padding_block: var(--space-m);--wrapper_max_width: 80rem;">  <div class="flex-block gap---space-m" data-mod style="--flex_block_gap: var(--space-m);">   <div class="[ cover-body ]">  <div class="[ input-group ]" data-astro-cid-ovtlrinh style="--width: 100%;">  <input placeholder="Type alliance name to find" x-model="filter" data-astro-cid-e5qsd3eu="true" class="astro-ovtlrinh" data-astro-cid-ovtlrinh style="--width: 100%;"> </div>   <div class="[ relative ]" data-astro-cid-e5qsd3eu> <div class="flex-block [ alliance-list ][ light-transparency w-full ] gap-0" data-astro-cid-e5qsd3eu="true" data-mod style="--flex_block_gap: 0;"> <button type="button" alliance-finder-action="accept" alliance-finder-data="{&#34;id&#34;:1,&#34;name&#34;:&#34;Minmatar Fleet Alliance&#34;,&#34;members_count&#34;:7,&#34;alliance_id&#34;:99011978}" x-show="show_item($el)" x-on:click="(filter = '')" data-astro-cid-e5qsd3eu> <div class="fixed-fluid" data-astro-cid-e5qsd3eu="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-e5qsd3eu> <img loading="lazy" src="https://images.evetech.net/alliances/99011978/logo?size=64" height="64" width="64" alt="Minmatar Fleet Alliance profile picture" data-astro-cid-e5qsd3eu> </picture> <div class="!flex !items-center" data-astro-cid-e5qsd3eu="true">  <div class="flex-block [ items-start ] gap-0" data-astro-cid-e5qsd3eu="true" data-mod style="--flex_block_gap: 0;">  <span class="[ truncate ]" data-astro-cid-e5qsd3eu>Minmatar Fleet Alliance</span> <small data-astro-cid-e5qsd3eu>7 members count</small>  </div>   </div>  </div>   </button><button type="button" alliance-finder-action="accept" alliance-finder-data="{&#34;id&#34;:2,&#34;name&#34;:&#34;Ushra'Khan&#34;,&#34;members_count&#34;:11,&#34;alliance_id&#34;:1966049571}" x-show="show_item($el)" x-on:click="(filter = '')" data-astro-cid-e5qsd3eu> <div class="fixed-fluid" data-astro-cid-e5qsd3eu="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-e5qsd3eu> <img loading="lazy" src="https://images.evetech.net/alliances/1966049571/logo?size=64" height="64" width="64" alt="Ushra'Khan profile picture" data-astro-cid-e5qsd3eu> </picture> <div class="!flex !items-center" data-astro-cid-e5qsd3eu="true">  <div class="flex-block [ items-start ] gap-0" data-astro-cid-e5qsd3eu="true" data-mod style="--flex_block_gap: 0;">  <span class="[ truncate ]" data-astro-cid-e5qsd3eu>Ushra&#39;Khan</span> <small data-astro-cid-e5qsd3eu>11 members count</small>  </div>   </div>  </div>   </button> </div>  </div>  </div>   </div>   </div>   </div>  <div x-bind:class="(alliances_finder_open ? 'open' : '')" alliance-finder-action="close" x-on:click="(filter = '')" data-astro-cid-e5qsd3eu></div> </div> "
`;
